<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Meet Your Faculty | Infectious Disease Genomic Epidemiology 2023</title>
  <meta name="description" content="Workshop website for the 2023 Infectious Disease Genomic Epidemiology Canadian Bioinformatics Workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Meet Your Faculty | Infectious Disease Genomic Epidemiology 2023" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://bioinformaticsdotca.github.io/IDE_2023/img/bioinformatics_logo.png" />
  <meta property="og:description" content="Workshop website for the 2023 Infectious Disease Genomic Epidemiology Canadian Bioinformatics Workshop" />
  <meta name="github-repo" content="bioinformaticsdotca/IDE_2023" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Meet Your Faculty | Infectious Disease Genomic Epidemiology 2023" />
  
  <meta name="twitter:description" content="Workshop website for the 2023 Infectious Disease Genomic Epidemiology Canadian Bioinformatics Workshop" />
  <meta name="twitter:image" content="https://bioinformaticsdotca.github.io/IDE_2023/img/bioinformatics_logo.png" />

<meta name="author" content="Faculty: William Hsiao, Aaron Petkau, Jared Simpson, Fiona Brinkman, Finlay Maguire, Guillaume Bourque, Gary Van Domselaar, Ed Taboada, Emma Griffiths, Andrew McArthur, Michelle Brazas, Rhiannon Cameron, Karyn Mukiri, Miguel Prieto, Jimmy Liu, Jalees Nasir, Madeline McCarthy, Charlie Barclay, Nia Hughes, and Zhibin Lu" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="img/favicon.ico" type="image/x-icon" />
<link rel="prev" href="index.html"/>
<link rel="next" href="data-and-compute-setup.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.1/css/all.min.css">



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://bioinformaticsdotca.github.io/">
 <img src="img/bioinformatics_logo.png" width="90%" alt="bioinformatics.ca logo" style="display:block; margin-left:auto; margin-right:auto; margin-top:15px;">
 </a>
<li><center><strong><a href="./"><br>Infectious Disease Genomic Epidemiology<br><br></a></strong></center></li>
<a href="./">
 <img src="img/CBW_epidemiology_icon.png" width="70%" alt="Infectious Disease Genomic Epidemiology logo" style="display:block; margin-left:auto; margin-right:auto; ">
 </a>
<br>

<li class="divider"></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Workshop Info</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#class-photo"><i class="fa fa-check"></i>Class Photo</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i>Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#pre-work"><i class="fa fa-check"></i>Pre-work</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="meet-your-faculty.html"><a href="meet-your-faculty.html"><i class="fa fa-check"></i>Meet Your Faculty</a></li>
<li class="chapter" data-level="" data-path="data-and-compute-setup.html"><a href="data-and-compute-setup.html"><i class="fa fa-check"></i>Data and Compute Setup</a></li>
<li class="part"><span><b>II Modules</b></span></li>
<li class="chapter" data-level="" data-path="module-1-introduction-to-genomic-epidemiology.html"><a href="module-1-introduction-to-genomic-epidemiology.html"><i class="fa fa-check"></i>Module 1: Introduction to Genomic Epidemiology</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-introduction-to-genomic-epidemiology.html"><a href="module-1-introduction-to-genomic-epidemiology.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2-phylogenetic-analysis.html"><a href="module-2-phylogenetic-analysis.html"><i class="fa fa-check"></i>Module 2: Phylogenetic Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-2-phylogenetic-analysis.html"><a href="module-2-phylogenetic-analysis.html#lecture-1"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-2-phylogenetic-analysis.html"><a href="module-2-phylogenetic-analysis.html#aws-setup"><i class="fa fa-check"></i>AWS Setup</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3-data-curation-and-data-sharing.html"><a href="module-3-data-curation-and-data-sharing.html"><i class="fa fa-check"></i>Module 3: Data Curation and Data Sharing</a>
<ul>
<li class="chapter" data-level="" data-path="module-3-data-curation-and-data-sharing.html"><a href="module-3-data-curation-and-data-sharing.html#lecture-2"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-3-data-curation-and-data-sharing.html"><a href="module-3-data-curation-and-data-sharing.html#lab"><i class="fa fa-check"></i>Lab</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4-viral-pathogen-genomic-analysis---single-nucleotide-variants.html"><a href="module-4-viral-pathogen-genomic-analysis---single-nucleotide-variants.html"><i class="fa fa-check"></i>Module 4: Viral Pathogen Genomic Analysis - Single Nucleotide Variants</a>
<ul>
<li class="chapter" data-level="" data-path="module-4-viral-pathogen-genomic-analysis---single-nucleotide-variants.html"><a href="module-4-viral-pathogen-genomic-analysis---single-nucleotide-variants.html#lecture-3"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-4-viral-pathogen-genomic-analysis---single-nucleotide-variants.html"><a href="module-4-viral-pathogen-genomic-analysis---single-nucleotide-variants.html#lab-1"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-4-viral-pathogen-genomic-analysis---single-nucleotide-variants.html"><a href="module-4-viral-pathogen-genomic-analysis---single-nucleotide-variants.html#virus-short-read-genome-assembly-and-variant-analysis"><i class="fa fa-check"></i>Virus Short-Read Genome Assembly and Variant Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html"><i class="fa fa-check"></i>Module 5: Bacterial Pathogen Genomic Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#lecture-4"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#lab-2"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#setup"><i class="fa fa-check"></i>Setup</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#background"><i class="fa fa-check"></i>Background</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#learning-objectives"><i class="fa fa-check"></i>Learning Objectives</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#analysis-dataset"><i class="fa fa-check"></i>Analysis Dataset</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#getting-started"><i class="fa fa-check"></i>Getting Started</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#cgmlst-qc"><i class="fa fa-check"></i>cgMLST QC</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#locus-quality"><i class="fa fa-check"></i>Locus Quality</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#defining-core-genes"><i class="fa fa-check"></i>Defining Core Genes</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#genome-quality"><i class="fa fa-check"></i>Genome Quality</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#hamming-distance"><i class="fa fa-check"></i>Hamming Distance</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#dendrogram-construction"><i class="fa fa-check"></i>Dendrogram Construction</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#a-simple-tree-vis"><i class="fa fa-check"></i>A Simple Tree Vis</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#clustering-by-distance"><i class="fa fa-check"></i>Clustering by Distance</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#cluster-analysis"><i class="fa fa-check"></i>Cluster Analysis</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#investigating-local-core-genes"><i class="fa fa-check"></i>Investigating Local Core Genes</a></li>
<li class="chapter" data-level="" data-path="module-5-bacterial-pathogen-genomic-analysis.html"><a href="module-5-bacterial-pathogen-genomic-analysis.html#group-exercise"><i class="fa fa-check"></i>Group Exercise</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-6-antimicrobial-resistant-gene-amr-analysis.html"><a href="module-6-antimicrobial-resistant-gene-amr-analysis.html"><i class="fa fa-check"></i>Module 6: Antimicrobial Resistant Gene (AMR) Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-6-antimicrobial-resistant-gene-amr-analysis.html"><a href="module-6-antimicrobial-resistant-gene-amr-analysis.html#lecture-5"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-6-antimicrobial-resistant-gene-amr-analysis.html"><a href="module-6-antimicrobial-resistant-gene-amr-analysis.html#lab-3"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-6-antimicrobial-resistant-gene-amr-analysis.html"><a href="module-6-antimicrobial-resistant-gene-amr-analysis.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="module-6-antimicrobial-resistant-gene-amr-analysis.html"><a href="module-6-antimicrobial-resistant-gene-amr-analysis.html#card-website-and-antibiotic-resistance-ontology"><i class="fa fa-check"></i>CARD Website and Antibiotic Resistance Ontology</a></li>
<li class="chapter" data-level="" data-path="module-6-antimicrobial-resistant-gene-amr-analysis.html"><a href="module-6-antimicrobial-resistant-gene-amr-analysis.html#rgi-for-genome-analysis"><i class="fa fa-check"></i>RGI for Genome Analysis</a></li>
<li class="chapter" data-level="" data-path="module-6-antimicrobial-resistant-gene-amr-analysis.html"><a href="module-6-antimicrobial-resistant-gene-amr-analysis.html#rgi-at-the-command-line"><i class="fa fa-check"></i>RGI at the Command Line</a></li>
<li class="chapter" data-level="" data-path="module-6-antimicrobial-resistant-gene-amr-analysis.html"><a href="module-6-antimicrobial-resistant-gene-amr-analysis.html#rgi-for-merged-metagenomic-reads"><i class="fa fa-check"></i>RGI for Merged Metagenomic Reads</a></li>
<li class="chapter" data-level="" data-path="module-6-antimicrobial-resistant-gene-amr-analysis.html"><a href="module-6-antimicrobial-resistant-gene-amr-analysis.html#metagenomic-sequencing-reads-and-the-kma-algorithm"><i class="fa fa-check"></i>Metagenomic Sequencing Reads and the KMA Algorithm</a></li>
<li class="chapter" data-level="" data-path="module-6-antimicrobial-resistant-gene-amr-analysis.html"><a href="module-6-antimicrobial-resistant-gene-amr-analysis.html#pathogen-of-origin-prediction"><i class="fa fa-check"></i>Pathogen of Origin Prediction</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-7-phylodynamics-and-transmission-dynamics.html"><a href="module-7-phylodynamics-and-transmission-dynamics.html"><i class="fa fa-check"></i>Module 7: Phylodynamics and Transmission Dynamics</a>
<ul>
<li class="chapter" data-level="" data-path="module-7-phylodynamics-and-transmission-dynamics.html"><a href="module-7-phylodynamics-and-transmission-dynamics.html#lecture-6"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-7-phylodynamics-and-transmission-dynamics.html"><a href="module-7-phylodynamics-and-transmission-dynamics.html#lab-4"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-7-phylodynamics-and-transmission-dynamics.html"><a href="module-7-phylodynamics-and-transmission-dynamics.html#introduction-2"><i class="fa fa-check"></i>1. Introduction</a></li>
<li class="chapter" data-level="" data-path="module-7-phylodynamics-and-transmission-dynamics.html"><a href="module-7-phylodynamics-and-transmission-dynamics.html#list-of-software-for-tutorial"><i class="fa fa-check"></i>2. List of Software for Tutorial</a></li>
<li class="chapter" data-level="" data-path="module-7-phylodynamics-and-transmission-dynamics.html"><a href="module-7-phylodynamics-and-transmission-dynamics.html#exercise-setup"><i class="fa fa-check"></i>3. Exercise Setup</a></li>
<li class="chapter" data-level="" data-path="module-7-phylodynamics-and-transmission-dynamics.html"><a href="module-7-phylodynamics-and-transmission-dynamics.html#phylodynamics-analysis"><i class="fa fa-check"></i>4. Phylodynamics Analysis</a></li>
<li class="chapter" data-level="" data-path="module-7-phylodynamics-and-transmission-dynamics.html"><a href="module-7-phylodynamics-and-transmission-dynamics.html#visualizing-the-phylodynamic-analysis-alongside-epidemiological-metadata"><i class="fa fa-check"></i>5. Visualizing the Phylodynamic Analysis Alongside Epidemiological Metadata</a></li>
<li class="chapter" data-level="" data-path="module-7-phylodynamics-and-transmission-dynamics.html"><a href="module-7-phylodynamics-and-transmission-dynamics.html#selection-analysis"><i class="fa fa-check"></i>6. Selection Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-8-emerging-pathogen-detection-and-identification.html"><a href="module-8-emerging-pathogen-detection-and-identification.html"><i class="fa fa-check"></i>Module 8: Emerging Pathogen Detection and Identification</a>
<ul>
<li class="chapter" data-level="" data-path="module-8-emerging-pathogen-detection-and-identification.html"><a href="module-8-emerging-pathogen-detection-and-identification.html#lecture-7"><i class="fa fa-check"></i>Lecture</a>
<ul>
<li class="chapter" data-level="" data-path="module-8-emerging-pathogen-detection-and-identification.html"><a href="module-8-emerging-pathogen-detection-and-identification.html#introduction-3"><i class="fa fa-check"></i>1. Introduction</a></li>
<li class="chapter" data-level="" data-path="module-8-emerging-pathogen-detection-and-identification.html"><a href="module-8-emerging-pathogen-detection-and-identification.html#list-fo-software-for-tutorial-and-its-respective-documentation"><i class="fa fa-check"></i>2. List fo Software for Tutorial and its Respective Documentation</a></li>
<li class="chapter" data-level="" data-path="module-8-emerging-pathogen-detection-and-identification.html"><a href="module-8-emerging-pathogen-detection-and-identification.html#exercise-setup-1"><i class="fa fa-check"></i>3. Exercise Setup</a></li>
<li class="chapter" data-level="" data-path="module-8-emerging-pathogen-detection-and-identification.html"><a href="module-8-emerging-pathogen-detection-and-identification.html#exercise"><i class="fa fa-check"></i>4. Exercise</a></li>
<li class="chapter" data-level="" data-path="module-8-emerging-pathogen-detection-and-identification.html"><a href="module-8-emerging-pathogen-detection-and-identification.html#final-words"><i class="fa fa-check"></i>5. Final words</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="keynote-lecture-follow-that-virus-genomics-leading-the-way.html"><a href="keynote-lecture-follow-that-virus-genomics-leading-the-way.html"><i class="fa fa-check"></i>Keynote Lecture | Follow That Virus!: Genomics Leading the Way</a>
<ul>
<li class="chapter" data-level="" data-path="keynote-lecture-follow-that-virus-genomics-leading-the-way.html"><a href="keynote-lecture-follow-that-virus-genomics-leading-the-way.html#lecture-8"><i class="fa fa-check"></i>Lecture</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-9-mobile-genetic-elements-and-environmental-microbiome.html"><a href="module-9-mobile-genetic-elements-and-environmental-microbiome.html"><i class="fa fa-check"></i>Module 9: Mobile Genetic Elements and Environmental Microbiome</a>
<ul>
<li class="chapter" data-level="" data-path="module-9-mobile-genetic-elements-and-environmental-microbiome.html"><a href="module-9-mobile-genetic-elements-and-environmental-microbiome.html#lecture-9"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-9-mobile-genetic-elements-and-environmental-microbiome.html"><a href="module-9-mobile-genetic-elements-and-environmental-microbiome.html#lab-5"><i class="fa fa-check"></i>Lab</a></li>
</ul></li>
<li class="divider"></li>
<h1 id="sponsors">Sponsors</h1>
<center>
<a href="https://genomecanada.ca/"><img src="img/sponsors/genomecanada.png" height="70" style="display:inline-block;"  style="padding-bottom: 20px" alt="Genome Canada logo.">
<a href="https://www.ontariogenomics.ca/"><img src="img/sponsors/ontariogenomics.png" height="70" style="display:inline-block;" style="padding-bottom: 20px;" alt="Ontario Genomics logo.">
<br><br>
<a href="https://oicr.on.ca/"><img src="img/sponsors/oicr.png" width=25% style="display:inline-block;" style="padding-bottom: 20px" alt="Ontario Institute for Cancer Research logo.">
<a href="https://www.hpcforhealth.ca/"><img src="img/sponsors/hpc4health.png" width=60% style="display:inline-block;" style="padding-bottom: 20px "alt="HPC4Health logo.">
<br><br>
<a href="https://www.cihr-irsc.gc.ca"><img src="img/sponsors/cihr.png" style="display:inline-block;" width=35% style="padding-bottom: 20px" alt="Canadian Institutes of Health Research logo."></a>

<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li></center>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Infectious Disease Genomic Epidemiology 2023</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="meet-your-faculty" class="section level1 hasAnchor">
<h1>Meet Your Faculty<a href="meet-your-faculty.html#meet-your-faculty" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="william-hsiao" class="section level4 hasAnchor">
<h4>William Hsiao<a href="meet-your-faculty.html#william-hsiao" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/william-hsiao.png" width="200" alt="Photo of William Hsiao"></p>
<blockquote>
<p>Associate Professor <br>
Faculty of Health Sciences, Simon Fraser University <br>
Vancouver, BC, CA</p>
<p>— <a href="mailto:wwhsiao@sfu.ca" class="email">wwhsiao@sfu.ca</a> <br>
<a href="www.cidgoh.ca">www.cidgoh.ca</a></p>
</blockquote>
<p>William Hsiao is a public health infectious disease researcher with a background in microbial
genomics and bioinformatics. He is an associate professor in the Faculty of Health Sciences at
Simon Fraser University and an affiliated researcher at BCCDC Public Health Laboratory and at
Canada’s Michael Smith Genome Sciences Centre. Will leads an interdisciplinary group of
researchers interested in solving practical public health and animal health problems through a
One Health lens at the Centre for Infectious Disease Genomics and One Health.</p>
</div>
<div id="aaron-petkau" class="section level4 hasAnchor">
<h4>Aaron Petkau<a href="meet-your-faculty.html#aaron-petkau" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/aaron-petkau.png" width="200" alt="Photo of Aaron Petkau"></p>
<blockquote>
<p>Head of Bioinformatics Pipeline Development <br>
Science Technology Cores and Services <br>
National Microbiology Laboratory <br>
Public Health Agency of Canada <br>
Winnipeg, MB, CA</p>
<p>— <a href="mailto:aaron.petkau@phac-aspc.gc.ca" class="email">aaron.petkau@phac-aspc.gc.ca</a> <br></p>
</blockquote>
<p>William Hsiao is a public health infectious disease researcher with a background in microbial
genomics and bioinformatics. He is an associate professor in the Faculty of Health Sciences at
Simon Fraser University and an affiliated researcher at BCCDC Public Health Laboratory and at
Canada’s Michael Smith Genome Sciences Centre. Will leads an interdisciplinary group of
researchers interested in solving practical public health and animal health problems through a
One Health lens at the Centre for Infectious Disease Genomics and One Health.</p>
</div>
<div id="jared-simpson" class="section level4 hasAnchor">
<h4>Jared Simpson<a href="meet-your-faculty.html#jared-simpson" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/jared-simpson.png" width="200" alt="Photo of Jared Simpson"></p>
<blockquote>
<p>Principal Investigator, Ontario Institute for Cancer Research <br>
Assistant Prof. Department of Computer Science, University of Toronto <br>
Vancouver, BC, CA</p>
<p>—<a href="https://simpsonlab.github.io/">https://simpsonlab.github.io/</a></p>
</blockquote>
<p>Dr. Simpson develops algorithms and software for the analysis of
high-throughput sequencing data. He is interested in de novo assembly and the detection of
sequence variation in individuals, cancers and populations, with a focus on long read
sequencing technologies. Dr. Simpson developed the ABYSS, SGA and nanopolish software
packages.</p>
</div>
<div id="fiona-brinkman" class="section level4 hasAnchor">
<h4>Fiona Brinkman<a href="meet-your-faculty.html#fiona-brinkman" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/fiona-brinkman.png" width="200" alt="Photo of Fiona Brinkman"></p>
<blockquote>
<p>Distinguished Professor, FRSC, Department of Molecular Biology and Biochemistry <br>
Associate Member, School of Computing Science and Faculty of Health Sciences <br>
Simon Fraser University <br>
Burnaby, BC, CA</p>
<p>— <a href="mailto:brinkman@sfu.ca" class="email">brinkman@sfu.ca</a> <br>
<a href="https://www.brinkmanlab.ca/">https://www.brinkmanlab.ca/</a></p>
</blockquote>
<p>Dr. Brinkman is developing bioinformatic resources to better track infectious diseases using
genomic data, and improve prediction of new vaccine/drug targets. Her primary aim is to
develop more sustainable, integrated approaches for infectious disease control, however she is
also applying her methods to aid allergy, child health, and environmental research.</p>
</div>
<div id="finlay-maguire" class="section level4 hasAnchor">
<h4>Finlay Maguire<a href="meet-your-faculty.html#finlay-maguire" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/finlay-maguire.png" width="200" alt="Photo of Finlay Maguire"></p>
<blockquote>
<p>Assistant Professor <br>
Faculty of Computer Science and Department of Community Health &amp; Epidemiology <br>
Dalhousie University <br>
Halifax, NS, CA</p>
<p>— <a href="mailto:finlay.maguire@dal.ca" class="email">finlay.maguire@dal.ca</a> <br>
<a href="finlaymagui.re">finlaymagui.re</a></p>
</blockquote>
<p>Finlay Maguire is a data scientist whose work centers on leveraging data in innovative ways to
answer questions related to applied health and social issues. This includes developing
bioinformatics methods to more effectively use genomic data to mitigate infectious diseases and
broad interdisciplinary collaborations in areas such as refugee healthcare provision and online
radicalisation. They are an active contributor to the national and international public health
responses to the COVID-19 pandemic and act as a genomic epidemiology advisor for
Sunnybrook’s Shared Hospital Laboratory.</p>
</div>
<div id="guillaume-bourque" class="section level4 hasAnchor">
<h4>Guillaume Bourque<a href="meet-your-faculty.html#guillaume-bourque" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/guillaume-bourque.png" width="200" alt="Photo of Guillaume Bourque"></p>
<blockquote>
<p>Professor, McGill University <br>
Director of Bioinformatics, Genome Quebec Innovation Centre <br>
Director, Canadian Centre of Computational Genomics <br>
Director, McGill initiative for Computational Medicine</p>
</blockquote>
<p>Dr. Bourque’s research interests are in comparative and functional genomics with a special
emphasis on applications of next-generation sequencing technologies. His lab develops
advanced tools and scalable computational infrastructure to enable large-scale applied research
projects.</p>
</div>
<div id="gary-van-domselaar" class="section level4 hasAnchor">
<h4>Gary Van Domselaar<a href="meet-your-faculty.html#gary-van-domselaar" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/gary-van-domselaar.png" width="200" alt="Photo of Guillaume Bourque"></p>
<blockquote>
<p>Chief Bioinformatics Scientist <br>
National Microbiology Laboratory <br>
Public Health Agency of Canada <br>
Winnipeg, MB, CA</p>
<p>— <a href="mailto:gary.vandomselaar@phac-aspc.gc.ca" class="email">gary.vandomselaar@phac-aspc.gc.ca</a></p>
</blockquote>
<p>Dr. Gary Van Domselaar, PhD (University of Alberta, 2003) is the Chief Bioinformatics Scientist at
the National Microbiology Laboratory in Winnipeg Canada and Associate Professor in the
Department of Medical Microbiology and Infectious Diseases at the University of Manitoba. Dr. Van
Domselaar’s lab develops bioinformatics methods and pipelines to understand, track, and control
circulating infectious diseases in Canada and globally. His research and development activities span
metagenomics, infectious disease genomic epidemiology, genome annotation, population structure
analysis, and microbial genome wide association studies. His lab contributes to large-scale national
and international genomics and bioinformatics efforts, including the Bioinformatics Workgroup of the
Canadian Genomics Research and Development Initiative Interdepartmental Project on
Antimicrobial Resistance, the Integrated Rapid Infectious Disease Analysis (IRIDA) project to
develop an integrated computational platform for infectious disease outbreak investigations, the
Canadian COVID-19 genomics network (CanCOGeN), and the Canadian Public Health Laboratory
Network COVID-19 Genomics Program. Dr. Van Domselaar serves on a number of national and
international scientific advisory groups, including the US Centers for Disease Control, the Global
Coalition for Science and Regulatory Research, and PHA4GE.</p>
</div>
<div id="ed-taboada" class="section level4 hasAnchor">
<h4>Ed Taboada<a href="meet-your-faculty.html#ed-taboada" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/ed-taboada.png" width="200" alt="Photo of Ed Taboada"></p>
<blockquote>
<p>Research Scientist <br>
Genomic Epidemiology Research Unit <br>
Division of Enteric Diseases <br>
National Microbiology Laboratory <br>
Public Health Agency of Canada <br>
Winnipeg, MB, CA</p>
<p>— <a href="mailto:eduardo.taboada@phac-aspc.gc.ca" class="email">eduardo.taboada@phac-aspc.gc.ca</a></p>
</blockquote>
<p>Eduardo Taboada is a research scientist in the Division of Enteric Diseases at the NML. Since
obtaining his PhD in molecular genetics at the Department of Biology, University of Ottawa in
1999, he has applied his expertise in molecular biology and population genetics towards the
development of comparative genomics and bioinformatics-based tools for studying various
aspects of the biology, ecology, and epidemiology of bacterial food- and waterborne pathogens,
with a focus on Campylobacter, Salmonella, and E. coli. Over these past two decades, Ed has
led or participated in several projects funded by Genome Canada and the federal government’s
Genomics Research and Development Initiative. His current research focus is on developing
quantitative tools for integrated genomic and epidemiologic analysis and the application of
analytical epidemiology approaches to genomic datasets.</p>
</div>
<div id="emma-griffiths" class="section level4 hasAnchor">
<h4>Emma Griffiths<a href="meet-your-faculty.html#emma-griffiths" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/emma-griffiths.png" width="200" alt="Photo of Emma Griffiths"></p>
<blockquote>
<p>Emma Griffiths <br>
Research Associate <br>
Faculty of Health Sciences, Simon Fraser University <br>
Vancouver, BC, Canada</p>
<p>— <a href="mailto:emma_griffiths@sfu.ca" class="email">emma_griffiths@sfu.ca</a></p>
</blockquote>
<p>Emma Griffiths is a research associate at the Centre for Infectious Disease Genomics and One
Health (CIDGOH) in the Faculty of Health Sciences at Simon Fraser University in Vancouver,
Canada. Her work focuses on developing and implementing ontologies and data standards for
public health and food safety genomics to help improve data harmonization and integration. She
is a member of the Standards Council of Canada and leads the Public Health Alliance for
Genomic Epidemiology (PHA4GE) Data Structures Working Group.</p>
</div>
<div id="andrew-mcarthur" class="section level4 hasAnchor">
<h4>Andrew McArthur<a href="meet-your-faculty.html#andrew-mcarthur" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/andrew-mcarthur.png" width="200" alt="Photo of Andrew McArthur"></p>
<blockquote>
<p>Professor and Director BDC Program <br>
Biochemistry and Biomedical Sciences <br>
McMaster University <br>
Hamilton, ON, CA</p>
</blockquote>
<p>Dr. McArthur is a Professor and David Braley Chair in Computational Biology at McMaster
University. Dr. McArthur has had a career in the United States and Canada, including
NIH-funded positions at the Marine Biological Laboratory and Brown University, where he led
the genome assembly of the diarrheal pathogen Giardia intestinalis, plus 10 years of experience
in the private sector. Dr. McArthur’s research team focuses on building tools, databases, and
algorithms for the genomic surveillance of infectious pathogens. He and his team developed the
Comprehensive Antibiotic Resistance Database (card.mcmaster.ca) and the SARS-CoV-2
Illumina GeNome Assembly Line software platform.</p>
</div>
<div id="michelle-brazas" class="section level4 hasAnchor">
<h4>Michelle Brazas<a href="meet-your-faculty.html#michelle-brazas" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/michelle-brazas.jpg" width="200" alt="Photo of Michelle Brazas"></p>
<blockquote>
<p>Acting Scientific Director <br>
Canadian Bioinformatics Workshops (CBW) <br>
Toronto, ON, CA</p>
<p>— <a href="mailto:support@bioinformatics.ca" class="email">support@bioinformatics.ca</a></p>
</blockquote>
<p>Dr. Michelle Brazas is the Associate Director for Adaptive Oncology at the Ontario Institute for
Cancer Research (OICR), and acting Scientific Director at Bioinformatics.ca. Previously, Dr.
Brazas was the Program Manager for Bioinformatics.ca and a faculty member in Biotechnology
at BCIT. Michelle co-founded and runs the Toronto Bioinformatics User Group (TorBUG) now in
its 11th season, and plays an active role in the International Society of Computational Biology
where she sits on the Board of Directors and Executive Board.</p>
</div>
<div id="rhiannon-cameron" class="section level4 hasAnchor">
<h4>Rhiannon Cameron<a href="meet-your-faculty.html#rhiannon-cameron" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/rhiannon-cameron.png" width="200" alt="Photo of Rhiannon Cameron"></p>
<blockquote>
<p>PhD Graduate Student Researcher <br>
Faculty of Health Sciences, Simon Fraser University <br>
Vancouver, BC, CA</p>
<p>— <a href="mailto:rcameron@sfu.ca" class="email">rcameron@sfu.ca</a> <br>
<a href="https://cmrn-rhi.github.io/">https://cmrn-rhi.github.io/</a></p>
</blockquote>
<p>Rhiannon completed her Bachelor of Science in Microbiology in 2019 and is currently a PhD
student in the Faculty of Health Sciences at SFU under the supervision of Dr. Hsiao at the
Centre for Infectious Disease Genomics and One Health (CIDGOH). Her work focuses on
ontology curation and development for outbreak investigation and surveillance of SARS-CoV-2
for the Canadian COVID-19 Genomics Network (CanCOGeN) and food-borne pathogen risk
assessment modeling for the Canadian Antimicrobial Resistance Genomics Research and
Development Initiative (AMR-GRDI), and developing ontology/metadata-curation training
resources. In her spare time, Rhiannon volunteers with the Vancouver Bioinformatics User
Group and the Neil Squire Society Computer Comforts program.</p>
</div>
<div id="karyn-mukiri" class="section level4 hasAnchor">
<h4>Karyn Mukiri<a href="meet-your-faculty.html#karyn-mukiri" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/karyn-mukiri.png" width="200" alt="Photo of Karyn Mukiri"></p>
<blockquote>
<p>PhD Graduate Student Researcher <br>
Faculty of Health Sciences, McMaster University <br>
Department of Biochemistry and Biomedical Sciences <br>
Hamilton, ON, CA</p>
<p>— <a href="mailto:mukirikm@mcmaster.ca" class="email">mukirikm@mcmaster.ca</a></p>
</blockquote>
<p>Karyn is a second-year PhD student under the supervision of Dr. Andrew McArthur where her
main research focus is on the development of predictive genomic annotation algorithms to
increase antibiotic resistance surveillance by the Resistance Gene Identifier. She is also a
McMaster alum, having completed her undergraduate degree in Biotechnology under the
Faculty of Engineering. Although the bulk of her work focuses on more algorithmic problems,
her current research efforts are in helping pinpoint instances of miscuration within the
Comprehensive Antibiotic Resistance Database.</p>
</div>
<div id="miguel-prieto" class="section level4 hasAnchor">
<h4>Miguel Prieto<a href="meet-your-faculty.html#miguel-prieto" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/miguel-prieto.png" width="200" alt="Photo of Miguel Prieto"></p>
<blockquote>
<p>PhD Graduate Student Researcher <br>
Simon Fraser University <br>
Faculty of Health Sciences <br>
Vancouver, BC, CA</p>
<p>— <a href="mailto:mprietog@sfu.ca" class="email">mprietog@sfu.ca</a></p>
</blockquote>
<p>Miguel is a first year PhD student studying the impact of human infectious diseases and the role
of the microbiome in humans. Particularly, he aims to use machine learning and metagenomic
analysis to predict poor outcomes in chronic respiratory diseases. Miguel is a Medical Doctor
from the Universidad del Valle in Colombia and has worked as a researcher and project
coordinator on translational studies of neglected tropical diseases like leishmaniasis. Before
coming to SFU, he completed a Master of Experimental Medicine at UBC exploring blood-based
gene expression biomarkers of nontuberculous mycobacterial infections in patients with cystic
fibrosis.</p>
</div>
<div id="jimmy-liu" class="section level4 hasAnchor">
<h4>Jimmy Liu<a href="meet-your-faculty.html#jimmy-liu" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/MISSINGHEADSHOT.png" width="200" alt="Photo of Jimmy Liu"></p>
<blockquote>
<p>PhD Graduate Student Researcher <br>
Simon Fraser University <br>
Department of Molecular Biology and Biochemistry <br>
Burnaby, BC, CA</p>
<p>— <a href="mailto:ccl40@sfu.ca" class="email">ccl40@sfu.ca</a></p>
</blockquote>
<p>Jimmy Liu completed his BSc. in Biochemistry at the University of British Columbia in 2019.
During his undergraduate study, he worked as a lab technologist at the JC Wilt Infectious
Diseases Research Centre where he developed passion for infectious disease genomics.
Inspired by the widespread adoption of genomic sequencing to inform decison making, he
decided to pursue further training in bioinformatics and genomic epidemiology by joining Dr.
William Hsiao’s group at the Simon Fraser University as a Ph.D. trainee. His thesis research
involves the joint application of machine learning, graph structures, and sequence alignment
algorithms to characterize the pan-genome evolution of Salmonella enterica. The discoveries
from his pan-genome analyses will drive the refinement of existing typing methods for disease
reporting and outbreak detection.</p>
</div>
<div id="jalees-nasir" class="section level4 hasAnchor">
<h4>Jalees Nasir<a href="meet-your-faculty.html#jalees-nasir" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/jalees-nasir.png" width="200" alt="Photo of Jalees Nasir"></p>
<blockquote>
<p>PhD Graduate Student Researcher <br>
McMaster University <br>
Department of Biochemistry and Biomedical Sciences <br>
Hamilton, ON, CA</p>
<p>— <a href="mailto:nasirja@mcmaster.ca" class="email">nasirja@mcmaster.ca</a></p>
</blockquote>
<p>Jalees Nasir is a Ph.D. Candidate in the McArthur Laboratory at McMaster University. His work
focuses on developing molecular epidemiological tools for the surveillance of respiratory viral
infections, including SARS-CoV-2. These tools primarily come in the form of molecular assays
for target enrichment, including bait capture. However, with the advent of the COVID-19
pandemic, he has also developed software for processing sequencing data (SARS-CoV-2
Illumina GeNome Assembly Line; SIGNAL) to contribute to ongoing surveillance efforts.</p>
</div>
<div id="madeline-mccarthy" class="section level4 hasAnchor">
<h4>Madeline McCarthy<a href="meet-your-faculty.html#madeline-mccarthy" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/MISSINGHEADSHOT.png" width="200" alt="Photo of Madeline McCarthy"></p>
<blockquote>
<p>PhD Graduate Student Researcher <br>
McMaster University <br>
Department of Biochemistry and Biomedical Sciences <br>
Hamilton, ON, CA</p>
<p>— <a href="mailto:mccarm15@mcmaster.ca" class="email">mccarm15@mcmaster.ca</a></p>
</blockquote>
<p>Madeline is a PhD Candidate in the McArthur Lab at McMaster University. Her work focuses on
the development of targeted, cost-effective methods for culture-free outbreak detection and
surveillance of bacterial pathogens. In addition to whole genome capture, she is testing the
feasibility of targeted, culture-free plasmid recovery using bait capture and long-read
sequencing. Prior to her PhD, she completed a MSc in microbiology at the University of
Saskatchewan where she studied E. coli outbreaks in broiler chickens before hanging up her
pipettes and switching to bioinformatics.</p>
</div>
<div id="charlie-barclay" class="section level4 hasAnchor">
<h4>Charlie Barclay<a href="meet-your-faculty.html#charlie-barclay" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/MISSINGHEADSHOT.png" width="200" alt="Photo of Charlie Barclay"></p>
<blockquote>
<p>MSc Graduate Student Researcher <br>
University of British Columbia <br>
Department of Physics and Astronomy <br>
Vancouver, BC, CA</p>
<p>— <a href="mailto:cbarcl01@mail.ubc.ca" class="email">cbarcl01@mail.ubc.ca</a></p>
</blockquote>
<p>Charlie completed her Masters in Marine Biology from the University of Southampton in 2013
and then spent 5 years working as a Data Manager at the Natural History Museum (NHM) in
London, during which she became involved in the Darwin Tree of Life project, standardizing
data at capture which led her to retrain in Bioinformatics. Charlie is currently working on the de
novo genome assembly of a comb jelly Mnemiopsis leidyi and how genomic insights of this
early branching metazoan can help us uncover mechanisms into cell differentiation and
multicellularity. Recently she has furthered her interest in FAIR data and ontologies through an
internship in Centre for Infectious Disease Genomics and One Health (CIDGOH).</p>
</div>
<div id="nia-hughes" class="section level4 hasAnchor">
<h4>Nia Hughes<a href="meet-your-faculty.html#nia-hughes" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/nia-hughes.jpeg" width="200" alt="Photo of Nia Hughes"></p>
<blockquote>
<p>Program Manager, Bioinformatics.ca <br>
Ontario Institute for Cancer Research <br>
Toronto, ON, CA</p>
<p>— <a href="mailto:nia.hughes@oicr.on.ca" class="email">nia.hughes@oicr.on.ca</a></p>
</blockquote>
<p>Nia is the Program Manager for Bioinformatics.ca, where she coordinates the Canadian
Bioinformatics Workshop Series. Prior to starting at OICR, she completed her M.Sc. in
Bioinformatics from the University of Guelph in 2020 before working there as a bioinformatician
studying epigenetic and transcriptomic patterns across maize varieties.</p>
</div>
<div id="zhibin-lu" class="section level4 hasAnchor">
<h4>Zhibin Lu<a href="meet-your-faculty.html#zhibin-lu" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><img src="img/faculty/MISSINGHEADSHOT.png" width="200" alt="Photo of Zhibin Lu"></p>
<blockquote>
<p>HPC and Bioinformatics Services Manager at Princess Margaret <br>
Cancer Centre, University Health Network <br>
Bioinformatics and HPC Core, UHN <br>
Toronto, ON, CA</p>
<p>— <a href="mailto:zhibin@gmail.com" class="email">zhibin@gmail.com</a> <br>
<a href="https://bhpc.uhnresearch.ca/">https://bhpc.uhnresearch.ca/</a></p>
</blockquote>
<p>Zhibin Lu is a senior manager at University Health Network Digital. He is responsible for UHN
HPC operations and scientific software. He manages two HPC clusters at UHN, including
system administration, user management, and maintenance of bioinformatics tools for
HPC4health. He is also skilled in Next-Gen sequence data analysis and has developed and
maintained bioinformatics pipelines at the Bioinformatics and HPC Core. He is a member of the
Digital Research Alliance of Canada Bioinformatics National Team and Scheduling National
Team.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="data-and-compute-setup.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/bioinformaticsdotca/IDE_2023/blob/main/001-faculty.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
